Enabling history cleanup on empty db fails Camunda startup

Not sure if this is because of Camunda or Micronaut Camunda, but when I configure history cleanup with batch window start time, Camunda startup fails with the stack trace below.

If I start Camunda once without history cleanup window batch window start time first, I can enable the setting and everything works as expected.

```
camunda:
  generic-properties:
    properties:
      history-cleanup-batch-window-start-time: "22:00"
```

```20:21:58.372 [main] ERROR org.camunda.bpm.engine.context - ENGINE-16004 Exception while closing command context: An exception occurred in the persistence layer. Please check the server logs for a detailed message and the entire exception stack trace.
org.camunda.bpm.engine.ProcessEngineException: An exception occurred in the persistence layer. Please check the server logs for a detailed message and the entire exception stack trace.
        at org.camunda.bpm.engine.impl.util.ExceptionUtil.wrapPersistenceException(ExceptionUtil.java:263)
        at org.camunda.bpm.engine.impl.util.ExceptionUtil.doWithExceptionWrapper(ExceptionUtil.java:257)
        at org.camunda.bpm.engine.impl.db.sql.DbSqlSession.selectById(DbSqlSession.java:120)
        at org.camunda.bpm.engine.impl.db.entitymanager.DbEntityManager.selectById(DbEntityManager.java:211)
        at org.camunda.bpm.engine.impl.persistence.entity.PropertyManager.findPropertyById(PropertyManager.java:29)
        at org.camunda.bpm.engine.impl.cmd.GetNextIdBlockCmd.execute(GetNextIdBlockCmd.java:40)
        at org.camunda.bpm.engine.impl.cmd.GetNextIdBlockCmd.execute(GetNextIdBlockCmd.java:28)
        at org.camunda.bpm.engine.impl.interceptor.CommandExecutorImpl.execute(CommandExecutorImpl.java:28)
        at org.camunda.bpm.engine.impl.interceptor.CommandContextInterceptor.execute(CommandContextInterceptor.java:110)
        at info.novatec.micronaut.camunda.bpm.feature.tx.MnTransactionInterceptor.lambda$execute$0(MnTransactionInterceptor.java:44)
        at io.micronaut.transaction.support.AbstractSynchronousTransactionManager.execute(AbstractSynchronousTransactionManager.java:142)
        at info.novatec.micronaut.camunda.bpm.feature.tx.MnTransactionInterceptor.execute(MnTransactionInterceptor.java:44)
        at org.camunda.bpm.engine.impl.interceptor.ProcessApplicationContextInterceptor.execute(ProcessApplicationContextInterceptor.java:70)
        at org.camunda.bpm.engine.impl.interceptor.CommandCounterInterceptor.execute(CommandCounterInterceptor.java:35)
        at org.camunda.bpm.engine.impl.interceptor.LogInterceptor.execute(LogInterceptor.java:33)
        at org.camunda.bpm.engine.impl.db.DbIdGenerator.getNewBlock(DbIdGenerator.java:49)
        at org.camunda.bpm.engine.impl.db.DbIdGenerator.getNextId(DbIdGenerator.java:41)
        at org.camunda.bpm.engine.impl.db.entitymanager.DbEntityManager.ensureHasId(DbEntityManager.java:643)
        at org.camunda.bpm.engine.impl.db.entitymanager.DbEntityManager.insert(DbEntityManager.java:525)
        at org.camunda.bpm.engine.impl.persistence.entity.JobManager.insertJob(JobManager.java:80)
        at org.camunda.bpm.engine.impl.persistence.entity.JobEntity.insert(JobEntity.java:165)
        at org.camunda.bpm.engine.impl.persistence.entity.JobManager.insertAndHintJobExecutor(JobManager.java:98)
        at org.camunda.bpm.engine.impl.cmd.HistoryCleanupCmd.createJobs(HistoryCleanupCmd.java:137)
        at org.camunda.bpm.engine.impl.cmd.HistoryCleanupCmd.execute(HistoryCleanupCmd.java:85)
        at org.camunda.bpm.engine.impl.cmd.HistoryCleanupCmd.execute(HistoryCleanupCmd.java:48)
        at org.camunda.bpm.engine.impl.interceptor.CommandExecutorImpl.execute(CommandExecutorImpl.java:28)
        at org.camunda.bpm.engine.impl.interceptor.CommandContextInterceptor.execute(CommandContextInterceptor.java:110)
        at info.novatec.micronaut.camunda.bpm.feature.tx.MnTransactionInterceptor.lambda$execute$0(MnTransactionInterceptor.java:44)
        at io.micronaut.transaction.support.AbstractSynchronousTransactionManager.execute(AbstractSynchronousTransactionManager.java:142)
        at info.novatec.micronaut.camunda.bpm.feature.tx.MnTransactionInterceptor.execute(MnTransactionInterceptor.java:44)
        at org.camunda.bpm.engine.impl.interceptor.ProcessApplicationContextInterceptor.execute(ProcessApplicationContextInterceptor.java:70)
        at org.camunda.bpm.engine.impl.interceptor.CommandCounterInterceptor.execute(CommandCounterInterceptor.java:35)
        at org.camunda.bpm.engine.impl.interceptor.LogInterceptor.execute(LogInterceptor.java:33)
        at org.camunda.bpm.engine.impl.HistoryServiceImpl.cleanUpHistoryAsync(HistoryServiceImpl.java:170)
        at org.camunda.bpm.engine.impl.HistoryServiceImpl.cleanUpHistoryAsync(HistoryServiceImpl.java:166)
        at org.camunda.bpm.engine.impl.BootstrapEngineCommand.createHistoryCleanupJob(BootstrapEngineCommand.java:85)
        at org.camunda.bpm.engine.impl.BootstrapEngineCommand.execute(BootstrapEngineCommand.java:57)
        at org.camunda.bpm.engine.impl.BootstrapEngineCommand.execute(BootstrapEngineCommand.java:41)
        at org.camunda.bpm.engine.impl.interceptor.CommandExecutorImpl.execute(CommandExecutorImpl.java:28)
        at org.camunda.bpm.engine.impl.interceptor.CommandContextInterceptor.execute(CommandContextInterceptor.java:110)
        at info.novatec.micronaut.camunda.bpm.feature.tx.MnTransactionInterceptor.lambda$execute$0(MnTransactionInterceptor.java:44)
        at io.micronaut.transaction.support.AbstractSynchronousTransactionManager.execute(AbstractSynchronousTransactionManager.java:142)
        at info.novatec.micronaut.camunda.bpm.feature.tx.MnTransactionInterceptor.execute(MnTransactionInterceptor.java:44)
        at org.camunda.bpm.engine.impl.interceptor.ProcessApplicationContextInterceptor.execute(ProcessApplicationContextInterceptor.java:70)
        at org.camunda.bpm.engine.impl.interceptor.CommandCounterInterceptor.execute(CommandCounterInterceptor.java:35)
        at org.camunda.bpm.engine.impl.interceptor.LogInterceptor.execute(LogInterceptor.java:33)
        at org.camunda.bpm.engine.impl.ProcessEngineImpl.executeSchemaOperations(ProcessEngineImpl.java:151)
        at org.camunda.bpm.engine.impl.ProcessEngineImpl.<init>(ProcessEngineImpl.java:110)
        at org.camunda.bpm.engine.impl.cfg.ProcessEngineConfigurationImpl.buildProcessEngine(ProcessEngineConfigurationImpl.java:1017)
        at info.novatec.micronaut.camunda.bpm.feature.ProcessEngineFactory.lambda$processEngine$0(ProcessEngineFactory.java:54)
        at io.micronaut.transaction.support.AbstractSynchronousTransactionManager.executeWrite(AbstractSynchronousTransactionManager.java:182)
        at info.novatec.micronaut.camunda.bpm.feature.ProcessEngineFactory.processEngine(ProcessEngineFactory.java:53)
        at info.novatec.micronaut.camunda.bpm.feature.$ProcessEngineFactory$ProcessEngine0$Definition.build(Unknown Source)
        at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2336)
        at io.micronaut.context.DefaultBeanContext.createAndRegisterSingletonInternal(DefaultBeanContext.java:3281)
        at io.micronaut.context.DefaultBeanContext.createAndRegisterSingleton(DefaultBeanContext.java:3267)
        at io.micronaut.context.DefaultBeanContext.getBeanForDefinition(DefaultBeanContext.java:2820)
        at io.micronaut.context.DefaultBeanContext.getBeanInternal(DefaultBeanContext.java:2782)
        at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1638)
        at io.micronaut.inject.provider.BeanProviderDefinition$1.get(BeanProviderDefinition.java:68)
        at info.novatec.micronaut.camunda.bpm.feature.initialization.InitProcessEngineService.onEvent(InitProcessEngineService.java:66)
        at info.novatec.micronaut.camunda.bpm.feature.initialization.$InitProcessEngineService$Definition$Exec.dispatch(Unknown Source)
        at io.micronaut.context.AbstractExecutableMethodsDefinition$DispatchedExecutableMethod.invoke(AbstractExecutableMethodsDefinition.java:378)
        at io.micronaut.context.DefaultBeanContext$BeanExecutionHandle.invoke(DefaultBeanContext.java:4027)
        at io.micronaut.aop.chain.AdapterIntroduction.intercept(AdapterIntroduction.java:83)
        at io.micronaut.aop.chain.MethodInterceptorChain.proceed(MethodInterceptorChain.java:137)
        at info.novatec.micronaut.camunda.bpm.feature.initialization.InitProcessEngineService$ApplicationEventListener$onEvent1$Intercepted.onApplicationEvent(Unknown Source)
        at io.micronaut.context.event.ApplicationEventPublisherFactory.notifyEventListeners(ApplicationEventPublisherFactory.java:262)
        at io.micronaut.context.event.ApplicationEventPublisherFactory.access$200(ApplicationEventPublisherFactory.java:60)
        at io.micronaut.context.event.ApplicationEventPublisherFactory$2.publishEvent(ApplicationEventPublisherFactory.java:229)
        at io.micronaut.context.DefaultBeanContext.publishEvent(DefaultBeanContext.java:1733)
        at io.micronaut.servlet.engine.server.AbstractServletServer.start(AbstractServletServer.java:80)
        at io.micronaut.servlet.engine.server.AbstractServletServer.start(AbstractServletServer.java:34)
        at io.micronaut.runtime.Micronaut.lambda$start$2(Micronaut.java:78)
        at java.base/java.util.Optional.ifPresent(Optional.java:178)
        at io.micronaut.runtime.Micronaut.start(Micronaut.java:76)
        at io.micronaut.runtime.Micronaut.run(Micronaut.java:320)
        at io.micronaut.runtime.Micronaut.run(Micronaut.java:306)
        at com.example.camunda.Application.main(Application.java:8)
Caused by: org.apache.ibatis.exceptions.PersistenceException: 
### Error querying database.  Cause: org.postgresql.util.PSQLException: ERROR: relation "act_ge_property" does not exist
  Position: 15
### The error may exist in org/camunda/bpm/engine/impl/mapping/entity/Property.xml
### The error may involve org.camunda.bpm.engine.impl.persistence.entity.PropertyEntity.selectProperty-Inline
### The error occurred while setting parameters
### SQL: select * from ACT_GE_PROPERTY where NAME_ = ?
### Cause: org.postgresql.util.PSQLException: ERROR: relation "act_ge_property" does not exist
  Position: 15
        at org.apache.ibatis.exceptions.ExceptionFactory.wrapException(ExceptionFactory.java:30)
        at org.apache.ibatis.session.defaults.DefaultSqlSession.selectList(DefaultSqlSession.java:149)
        at org.apache.ibatis.session.defaults.DefaultSqlSession.selectList(DefaultSqlSession.java:140)
        at org.apache.ibatis.session.defaults.DefaultSqlSession.selectOne(DefaultSqlSession.java:76)
        at org.camunda.bpm.engine.impl.db.sql.DbSqlSession.lambda$selectById$2(DbSqlSession.java:120)
        at org.camunda.bpm.engine.impl.util.ExceptionUtil.doWithExceptionWrapper(ExceptionUtil.java:254)
        ... 77 common frames omitted
Caused by: org.postgresql.util.PSQLException: ERROR: relation "act_ge_property" does not exist
  Position: 15
        at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2674)
        at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2364)
        at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:354)
        at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:484)
        at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:404)
        at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:162)
        at org.postgresql.jdbc.PgPreparedStatement.execute(PgPreparedStatement.java:151)
        at com.zaxxer.hikari.pool.ProxyPreparedStatement.execute(ProxyPreparedStatement.java:44)
        at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.execute(HikariProxyPreparedStatement.java)
        at org.apache.ibatis.executor.statement.PreparedStatementHandler.query(PreparedStatementHandler.java:64)
        at org.apache.ibatis.executor.statement.RoutingStatementHandler.query(RoutingStatementHandler.java:79)
        at org.apache.ibatis.executor.BatchExecutor.doQuery(BatchExecutor.java:92)
        at org.apache.ibatis.executor.BaseExecutor.queryFromDatabase(BaseExecutor.java:325)
        at org.apache.ibatis.executor.BaseExecutor.query(BaseExecutor.java:156)
        at org.apache.ibatis.executor.CachingExecutor.query(CachingExecutor.java:109)
        at org.apache.ibatis.executor.CachingExecutor.query(CachingExecutor.java:89)
        at org.apache.ibatis.session.defaults.DefaultSqlSession.selectList(DefaultSqlSession.java:147)
        ... 81 common frames omitted
20:21:58.376 [main] ERROR org.camunda.bpm.engine.context - ENGINE-16004 Exception while closing command context: An exception occurred in the persistence layer. Please check the server logs for a detailed message and the entire exception stack trace.
org.camunda.bpm.engine.ProcessEngineException: An exception occurred in the persistence layer. Please check the server logs for a detailed message and the entire exception stack trace.
        at org.camunda.bpm.engine.impl.util.ExceptionUtil.wrapPersistenceException(ExceptionUtil.java:263)
        at org.camunda.bpm.engine.impl.util.ExceptionUtil.doWithExceptionWrapper(ExceptionUtil.java:257)
        at org.camunda.bpm.engine.impl.db.sql.DbSqlSession.selectById(DbSqlSession.java:120)
        at org.camunda.bpm.engine.impl.db.entitymanager.DbEntityManager.selectById(DbEntityManager.java:211)
        at org.camunda.bpm.engine.impl.persistence.entity.PropertyManager.findPropertyById(PropertyManager.java:29)
        at org.camunda.bpm.engine.impl.cmd.GetNextIdBlockCmd.execute(GetNextIdBlockCmd.java:40)
        at org.camunda.bpm.engine.impl.cmd.GetNextIdBlockCmd.execute(GetNextIdBlockCmd.java:28)
        at org.camunda.bpm.engine.impl.interceptor.CommandExecutorImpl.execute(CommandExecutorImpl.java:28)
        at org.camunda.bpm.engine.impl.interceptor.CommandContextInterceptor.execute(CommandContextInterceptor.java:110)
        at info.novatec.micronaut.camunda.bpm.feature.tx.MnTransactionInterceptor.lambda$execute$0(MnTransactionInterceptor.java:44)
        at io.micronaut.transaction.support.AbstractSynchronousTransactionManager.execute(AbstractSynchronousTransactionManager.java:142)
        at info.novatec.micronaut.camunda.bpm.feature.tx.MnTransactionInterceptor.execute(MnTransactionInterceptor.java:44)
        at org.camunda.bpm.engine.impl.interceptor.ProcessApplicationContextInterceptor.execute(ProcessApplicationContextInterceptor.java:70)
        at org.camunda.bpm.engine.impl.interceptor.CommandCounterInterceptor.execute(CommandCounterInterceptor.java:35)
        at org.camunda.bpm.engine.impl.interceptor.LogInterceptor.execute(LogInterceptor.java:33)
        at org.camunda.bpm.engine.impl.db.DbIdGenerator.getNewBlock(DbIdGenerator.java:49)
        at org.camunda.bpm.engine.impl.db.DbIdGenerator.getNextId(DbIdGenerator.java:41)
        at org.camunda.bpm.engine.impl.db.entitymanager.DbEntityManager.ensureHasId(DbEntityManager.java:643)
        at org.camunda.bpm.engine.impl.db.entitymanager.DbEntityManager.insert(DbEntityManager.java:525)
        at org.camunda.bpm.engine.impl.persistence.entity.JobManager.insertJob(JobManager.java:80)
        at org.camunda.bpm.engine.impl.persistence.entity.JobEntity.insert(JobEntity.java:165)
        at org.camunda.bpm.engine.impl.persistence.entity.JobManager.insertAndHintJobExecutor(JobManager.java:98)
        at org.camunda.bpm.engine.impl.cmd.HistoryCleanupCmd.createJobs(HistoryCleanupCmd.java:137)
        at org.camunda.bpm.engine.impl.cmd.HistoryCleanupCmd.execute(HistoryCleanupCmd.java:85)
        at org.camunda.bpm.engine.impl.cmd.HistoryCleanupCmd.execute(HistoryCleanupCmd.java:48)
        at org.camunda.bpm.engine.impl.interceptor.CommandExecutorImpl.execute(CommandExecutorImpl.java:28)
        at org.camunda.bpm.engine.impl.interceptor.CommandContextInterceptor.execute(CommandContextInterceptor.java:110)
        at info.novatec.micronaut.camunda.bpm.feature.tx.MnTransactionInterceptor.lambda$execute$0(MnTransactionInterceptor.java:44)
        at io.micronaut.transaction.support.AbstractSynchronousTransactionManager.execute(AbstractSynchronousTransactionManager.java:142)
        at info.novatec.micronaut.camunda.bpm.feature.tx.MnTransactionInterceptor.execute(MnTransactionInterceptor.java:44)
        at org.camunda.bpm.engine.impl.interceptor.ProcessApplicationContextInterceptor.execute(ProcessApplicationContextInterceptor.java:70)
        at org.camunda.bpm.engine.impl.interceptor.CommandCounterInterceptor.execute(CommandCounterInterceptor.java:35)
        at org.camunda.bpm.engine.impl.interceptor.LogInterceptor.execute(LogInterceptor.java:33)
        at org.camunda.bpm.engine.impl.HistoryServiceImpl.cleanUpHistoryAsync(HistoryServiceImpl.java:170)
        at org.camunda.bpm.engine.impl.HistoryServiceImpl.cleanUpHistoryAsync(HistoryServiceImpl.java:166)
        at org.camunda.bpm.engine.impl.BootstrapEngineCommand.createHistoryCleanupJob(BootstrapEngineCommand.java:85)
        at org.camunda.bpm.engine.impl.BootstrapEngineCommand.execute(BootstrapEngineCommand.java:57)
        at org.camunda.bpm.engine.impl.BootstrapEngineCommand.execute(BootstrapEngineCommand.java:41)
        at org.camunda.bpm.engine.impl.interceptor.CommandExecutorImpl.execute(CommandExecutorImpl.java:28)
        at org.camunda.bpm.engine.impl.interceptor.CommandContextInterceptor.execute(CommandContextInterceptor.java:110)
        at info.novatec.micronaut.camunda.bpm.feature.tx.MnTransactionInterceptor.lambda$execute$0(MnTransactionInterceptor.java:44)
        at io.micronaut.transaction.support.AbstractSynchronousTransactionManager.execute(AbstractSynchronousTransactionManager.java:142)
        at info.novatec.micronaut.camunda.bpm.feature.tx.MnTransactionInterceptor.execute(MnTransactionInterceptor.java:44)
        at org.camunda.bpm.engine.impl.interceptor.ProcessApplicationContextInterceptor.execute(ProcessApplicationContextInterceptor.java:70)
        at org.camunda.bpm.engine.impl.interceptor.CommandCounterInterceptor.execute(CommandCounterInterceptor.java:35)
        at org.camunda.bpm.engine.impl.interceptor.LogInterceptor.execute(LogInterceptor.java:33)
        at org.camunda.bpm.engine.impl.ProcessEngineImpl.executeSchemaOperations(ProcessEngineImpl.java:151)
        at org.camunda.bpm.engine.impl.ProcessEngineImpl.<init>(ProcessEngineImpl.java:110)
        at org.camunda.bpm.engine.impl.cfg.ProcessEngineConfigurationImpl.buildProcessEngine(ProcessEngineConfigurationImpl.java:1017)
        at info.novatec.micronaut.camunda.bpm.feature.ProcessEngineFactory.lambda$processEngine$0(ProcessEngineFactory.java:54)
        at io.micronaut.transaction.support.AbstractSynchronousTransactionManager.executeWrite(AbstractSynchronousTransactionManager.java:182)
        at info.novatec.micronaut.camunda.bpm.feature.ProcessEngineFactory.processEngine(ProcessEngineFactory.java:53)
        at info.novatec.micronaut.camunda.bpm.feature.$ProcessEngineFactory$ProcessEngine0$Definition.build(Unknown Source)
        at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2336)
        at io.micronaut.context.DefaultBeanContext.createAndRegisterSingletonInternal(DefaultBeanContext.java:3281)
        at io.micronaut.context.DefaultBeanContext.createAndRegisterSingleton(DefaultBeanContext.java:3267)
        at io.micronaut.context.DefaultBeanContext.getBeanForDefinition(DefaultBeanContext.java:2820)
        at io.micronaut.context.DefaultBeanContext.getBeanInternal(DefaultBeanContext.java:2782)
        at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1638)
        at io.micronaut.inject.provider.BeanProviderDefinition$1.get(BeanProviderDefinition.java:68)
        at info.novatec.micronaut.camunda.bpm.feature.initialization.InitProcessEngineService.onEvent(InitProcessEngineService.java:66)
        at info.novatec.micronaut.camunda.bpm.feature.initialization.$InitProcessEngineService$Definition$Exec.dispatch(Unknown Source)
        at io.micronaut.context.AbstractExecutableMethodsDefinition$DispatchedExecutableMethod.invoke(AbstractExecutableMethodsDefinition.java:378)
        at io.micronaut.context.DefaultBeanContext$BeanExecutionHandle.invoke(DefaultBeanContext.java:4027)
        at io.micronaut.aop.chain.AdapterIntroduction.intercept(AdapterIntroduction.java:83)
        at io.micronaut.aop.chain.MethodInterceptorChain.proceed(MethodInterceptorChain.java:137)
        at info.novatec.micronaut.camunda.bpm.feature.initialization.InitProcessEngineService$ApplicationEventListener$onEvent1$Intercepted.onApplicationEvent(Unknown Source)
        at io.micronaut.context.event.ApplicationEventPublisherFactory.notifyEventListeners(ApplicationEventPublisherFactory.java:262)
        at io.micronaut.context.event.ApplicationEventPublisherFactory.access$200(ApplicationEventPublisherFactory.java:60)
        at io.micronaut.context.event.ApplicationEventPublisherFactory$2.publishEvent(ApplicationEventPublisherFactory.java:229)
        at io.micronaut.context.DefaultBeanContext.publishEvent(DefaultBeanContext.java:1733)
        at io.micronaut.servlet.engine.server.AbstractServletServer.start(AbstractServletServer.java:80)
        at io.micronaut.servlet.engine.server.AbstractServletServer.start(AbstractServletServer.java:34)
        at io.micronaut.runtime.Micronaut.lambda$start$2(Micronaut.java:78)
        at java.base/java.util.Optional.ifPresent(Optional.java:178)
        at io.micronaut.runtime.Micronaut.start(Micronaut.java:76)
        at io.micronaut.runtime.Micronaut.run(Micronaut.java:320)
        at io.micronaut.runtime.Micronaut.run(Micronaut.java:306)
        at com.example.camunda.Application.main(Application.java:8)
Caused by: org.apache.ibatis.exceptions.PersistenceException: 
### Error querying database.  Cause: org.postgresql.util.PSQLException: ERROR: relation "act_ge_property" does not exist
  Position: 15
### The error may exist in org/camunda/bpm/engine/impl/mapping/entity/Property.xml
### The error may involve org.camunda.bpm.engine.impl.persistence.entity.PropertyEntity.selectProperty-Inline
### The error occurred while setting parameters
### SQL: select * from ACT_GE_PROPERTY where NAME_ = ?
### Cause: org.postgresql.util.PSQLException: ERROR: relation "act_ge_property" does not exist
  Position: 15
        at org.apache.ibatis.exceptions.ExceptionFactory.wrapException(ExceptionFactory.java:30)
        at org.apache.ibatis.session.defaults.DefaultSqlSession.selectList(DefaultSqlSession.java:149)
        at org.apache.ibatis.session.defaults.DefaultSqlSession.selectList(DefaultSqlSession.java:140)
        at org.apache.ibatis.session.defaults.DefaultSqlSession.selectOne(DefaultSqlSession.java:76)
        at org.camunda.bpm.engine.impl.db.sql.DbSqlSession.lambda$selectById$2(DbSqlSession.java:120)
        at org.camunda.bpm.engine.impl.util.ExceptionUtil.doWithExceptionWrapper(ExceptionUtil.java:254)
        ... 77 common frames omitted
Caused by: org.postgresql.util.PSQLException: ERROR: relation "act_ge_property" does not exist
  Position: 15
        at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2674)
        at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2364)
        at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:354)
        at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:484)
        at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:404)
        at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:162)
        at org.postgresql.jdbc.PgPreparedStatement.execute(PgPreparedStatement.java:151)
        at com.zaxxer.hikari.pool.ProxyPreparedStatement.execute(ProxyPreparedStatement.java:44)
        at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.execute(HikariProxyPreparedStatement.java)
        at org.apache.ibatis.executor.statement.PreparedStatementHandler.query(PreparedStatementHandler.java:64)
        at org.apache.ibatis.executor.statement.RoutingStatementHandler.query(RoutingStatementHandler.java:79)
        at org.apache.ibatis.executor.BatchExecutor.doQuery(BatchExecutor.java:92)
        at org.apache.ibatis.executor.BaseExecutor.queryFromDatabase(BaseExecutor.java:325)
        at org.apache.ibatis.executor.BaseExecutor.query(BaseExecutor.java:156)
        at org.apache.ibatis.executor.CachingExecutor.query(CachingExecutor.java:109)
        at org.apache.ibatis.executor.CachingExecutor.query(CachingExecutor.java:89)
        at org.apache.ibatis.session.defaults.DefaultSqlSession.selectList(DefaultSqlSession.java:147)
        ... 81 common frames omitted
20:21:58.401 [main] ERROR io.micronaut.runtime.Micronaut - Error starting Micronaut server: Error starting HTTP server: Error instantiating bean of type  [org.camunda.bpm.engine.ProcessEngine]

Message: An exception occurred in the persistence layer. Please check the server logs for a detailed message and the entire exception stack trace.
Path Taken: ProcessEngine.processEngine(ProcessEngineConfiguration processEngineConfiguration,CamundaVersion camundaVersion,SynchronousTransactionManager transactionManager,BasicJdbcConfiguration basicJdbcConfiguration,ParallelInitializationService parallelInitializationService)
io.micronaut.http.server.exceptions.HttpServerException: Error starting HTTP server: Error instantiating bean of type  [org.camunda.bpm.engine.ProcessEngine]

Message: An exception occurred in the persistence layer. Please check the server logs for a detailed message and the entire exception stack trace.
Path Taken: ProcessEngine.processEngine(ProcessEngineConfiguration processEngineConfiguration,CamundaVersion camundaVersion,SynchronousTransactionManager transactionManager,BasicJdbcConfiguration basicJdbcConfiguration,ParallelInitializationService parallelInitializationService)
        at io.micronaut.servlet.engine.server.AbstractServletServer.start(AbstractServletServer.java:92)
        at io.micronaut.servlet.engine.server.AbstractServletServer.start(AbstractServletServer.java:34)
        at io.micronaut.runtime.Micronaut.lambda$start$2(Micronaut.java:78)
        at java.base/java.util.Optional.ifPresent(Optional.java:178)
        at io.micronaut.runtime.Micronaut.start(Micronaut.java:76)
        at io.micronaut.runtime.Micronaut.run(Micronaut.java:320)
        at io.micronaut.runtime.Micronaut.run(Micronaut.java:306)
        at com.example.camunda.Application.main(Application.java:8)
Caused by: io.micronaut.context.exceptions.BeanInstantiationException: Error instantiating bean of type  [org.camunda.bpm.engine.ProcessEngine]

Message: An exception occurred in the persistence layer. Please check the server logs for a detailed message and the entire exception stack trace.
Path Taken: ProcessEngine.processEngine(ProcessEngineConfiguration processEngineConfiguration,CamundaVersion camundaVersion,SynchronousTransactionManager transactionManager,BasicJdbcConfiguration basicJdbcConfiguration,ParallelInitializationService parallelInitializationService)
        at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2363)
        at io.micronaut.context.DefaultBeanContext.createAndRegisterSingletonInternal(DefaultBeanContext.java:3281)
        at io.micronaut.context.DefaultBeanContext.createAndRegisterSingleton(DefaultBeanContext.java:3267)
        at io.micronaut.context.DefaultBeanContext.getBeanForDefinition(DefaultBeanContext.java:2820)
        at io.micronaut.context.DefaultBeanContext.getBeanInternal(DefaultBeanContext.java:2782)
        at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1638)
        at io.micronaut.inject.provider.BeanProviderDefinition$1.get(BeanProviderDefinition.java:68)
        at info.novatec.micronaut.camunda.bpm.feature.initialization.InitProcessEngineService.onEvent(InitProcessEngineService.java:66)
        at info.novatec.micronaut.camunda.bpm.feature.initialization.$InitProcessEngineService$Definition$Exec.dispatch(Unknown Source)
        at io.micronaut.context.AbstractExecutableMethodsDefinition$DispatchedExecutableMethod.invoke(AbstractExecutableMethodsDefinition.java:378)
        at io.micronaut.context.DefaultBeanContext$BeanExecutionHandle.invoke(DefaultBeanContext.java:4027)
        at io.micronaut.aop.chain.AdapterIntroduction.intercept(AdapterIntroduction.java:83)
        at io.micronaut.aop.chain.MethodInterceptorChain.proceed(MethodInterceptorChain.java:137)
        at info.novatec.micronaut.camunda.bpm.feature.initialization.InitProcessEngineService$ApplicationEventListener$onEvent1$Intercepted.onApplicationEvent(Unknown Source)
        at io.micronaut.context.event.ApplicationEventPublisherFactory.notifyEventListeners(ApplicationEventPublisherFactory.java:262)
        at io.micronaut.context.event.ApplicationEventPublisherFactory.access$200(ApplicationEventPublisherFactory.java:60)
        at io.micronaut.context.event.ApplicationEventPublisherFactory$2.publishEvent(ApplicationEventPublisherFactory.java:229)
        at io.micronaut.context.DefaultBeanContext.publishEvent(DefaultBeanContext.java:1733)
        at io.micronaut.servlet.engine.server.AbstractServletServer.start(AbstractServletServer.java:80)
        ... 7 common frames omitted
Caused by: org.camunda.bpm.engine.ProcessEngineException: An exception occurred in the persistence layer. Please check the server logs for a detailed message and the entire exception stack trace.
        at org.camunda.bpm.engine.impl.util.ExceptionUtil.wrapPersistenceException(ExceptionUtil.java:263)
        at org.camunda.bpm.engine.impl.util.ExceptionUtil.doWithExceptionWrapper(ExceptionUtil.java:257)
        at org.camunda.bpm.engine.impl.db.sql.DbSqlSession.selectById(DbSqlSession.java:120)
        at org.camunda.bpm.engine.impl.db.entitymanager.DbEntityManager.selectById(DbEntityManager.java:211)
        at org.camunda.bpm.engine.impl.persistence.entity.PropertyManager.findPropertyById(PropertyManager.java:29)
        at org.camunda.bpm.engine.impl.cmd.GetNextIdBlockCmd.execute(GetNextIdBlockCmd.java:40)
        at org.camunda.bpm.engine.impl.cmd.GetNextIdBlockCmd.execute(GetNextIdBlockCmd.java:28)
        at org.camunda.bpm.engine.impl.interceptor.CommandExecutorImpl.execute(CommandExecutorImpl.java:28)
        at org.camunda.bpm.engine.impl.interceptor.CommandContextInterceptor.execute(CommandContextInterceptor.java:110)
        at info.novatec.micronaut.camunda.bpm.feature.tx.MnTransactionInterceptor.lambda$execute$0(MnTransactionInterceptor.java:44)
        at io.micronaut.transaction.support.AbstractSynchronousTransactionManager.execute(AbstractSynchronousTransactionManager.java:142)
        at info.novatec.micronaut.camunda.bpm.feature.tx.MnTransactionInterceptor.execute(MnTransactionInterceptor.java:44)
        at org.camunda.bpm.engine.impl.interceptor.ProcessApplicationContextInterceptor.execute(ProcessApplicationContextInterceptor.java:70)
        at org.camunda.bpm.engine.impl.interceptor.CommandCounterInterceptor.execute(CommandCounterInterceptor.java:35)
        at org.camunda.bpm.engine.impl.interceptor.LogInterceptor.execute(LogInterceptor.java:33)
        at org.camunda.bpm.engine.impl.db.DbIdGenerator.getNewBlock(DbIdGenerator.java:49)
        at org.camunda.bpm.engine.impl.db.DbIdGenerator.getNextId(DbIdGenerator.java:41)
        at org.camunda.bpm.engine.impl.db.entitymanager.DbEntityManager.ensureHasId(DbEntityManager.java:643)
        at org.camunda.bpm.engine.impl.db.entitymanager.DbEntityManager.insert(DbEntityManager.java:525)
        at org.camunda.bpm.engine.impl.persistence.entity.JobManager.insertJob(JobManager.java:80)
        at org.camunda.bpm.engine.impl.persistence.entity.JobEntity.insert(JobEntity.java:165)
        at org.camunda.bpm.engine.impl.persistence.entity.JobManager.insertAndHintJobExecutor(JobManager.java:98)
        at org.camunda.bpm.engine.impl.cmd.HistoryCleanupCmd.createJobs(HistoryCleanupCmd.java:137)
        at org.camunda.bpm.engine.impl.cmd.HistoryCleanupCmd.execute(HistoryCleanupCmd.java:85)
        at org.camunda.bpm.engine.impl.cmd.HistoryCleanupCmd.execute(HistoryCleanupCmd.java:48)
        at org.camunda.bpm.engine.impl.interceptor.CommandExecutorImpl.execute(CommandExecutorImpl.java:28)
        at org.camunda.bpm.engine.impl.interceptor.CommandContextInterceptor.execute(CommandContextInterceptor.java:110)
        at info.novatec.micronaut.camunda.bpm.feature.tx.MnTransactionInterceptor.lambda$execute$0(MnTransactionInterceptor.java:44)
        at io.micronaut.transaction.support.AbstractSynchronousTransactionManager.execute(AbstractSynchronousTransactionManager.java:142)
        at info.novatec.micronaut.camunda.bpm.feature.tx.MnTransactionInterceptor.execute(MnTransactionInterceptor.java:44)
        at org.camunda.bpm.engine.impl.interceptor.ProcessApplicationContextInterceptor.execute(ProcessApplicationContextInterceptor.java:70)
        at org.camunda.bpm.engine.impl.interceptor.CommandCounterInterceptor.execute(CommandCounterInterceptor.java:35)
        at org.camunda.bpm.engine.impl.interceptor.LogInterceptor.execute(LogInterceptor.java:33)
        at org.camunda.bpm.engine.impl.HistoryServiceImpl.cleanUpHistoryAsync(HistoryServiceImpl.java:170)
        at org.camunda.bpm.engine.impl.HistoryServiceImpl.cleanUpHistoryAsync(HistoryServiceImpl.java:166)
        at org.camunda.bpm.engine.impl.BootstrapEngineCommand.createHistoryCleanupJob(BootstrapEngineCommand.java:85)
        at org.camunda.bpm.engine.impl.BootstrapEngineCommand.execute(BootstrapEngineCommand.java:57)
        at org.camunda.bpm.engine.impl.BootstrapEngineCommand.execute(BootstrapEngineCommand.java:41)
        at org.camunda.bpm.engine.impl.interceptor.CommandExecutorImpl.execute(CommandExecutorImpl.java:28)
        at org.camunda.bpm.engine.impl.interceptor.CommandContextInterceptor.execute(CommandContextInterceptor.java:110)
        at info.novatec.micronaut.camunda.bpm.feature.tx.MnTransactionInterceptor.lambda$execute$0(MnTransactionInterceptor.java:44)
        at io.micronaut.transaction.support.AbstractSynchronousTransactionManager.execute(AbstractSynchronousTransactionManager.java:142)
        at info.novatec.micronaut.camunda.bpm.feature.tx.MnTransactionInterceptor.execute(MnTransactionInterceptor.java:44)
        at org.camunda.bpm.engine.impl.interceptor.ProcessApplicationContextInterceptor.execute(ProcessApplicationContextInterceptor.java:70)
        at org.camunda.bpm.engine.impl.interceptor.CommandCounterInterceptor.execute(CommandCounterInterceptor.java:35)
        at org.camunda.bpm.engine.impl.interceptor.LogInterceptor.execute(LogInterceptor.java:33)
        at org.camunda.bpm.engine.impl.ProcessEngineImpl.executeSchemaOperations(ProcessEngineImpl.java:151)
        at org.camunda.bpm.engine.impl.ProcessEngineImpl.<init>(ProcessEngineImpl.java:110)
        at org.camunda.bpm.engine.impl.cfg.ProcessEngineConfigurationImpl.buildProcessEngine(ProcessEngineConfigurationImpl.java:1017)
        at info.novatec.micronaut.camunda.bpm.feature.ProcessEngineFactory.lambda$processEngine$0(ProcessEngineFactory.java:54)
        at io.micronaut.transaction.support.AbstractSynchronousTransactionManager.executeWrite(AbstractSynchronousTransactionManager.java:182)
        at info.novatec.micronaut.camunda.bpm.feature.ProcessEngineFactory.processEngine(ProcessEngineFactory.java:53)
        at info.novatec.micronaut.camunda.bpm.feature.$ProcessEngineFactory$ProcessEngine0$Definition.build(Unknown Source)
        at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2336)
        ... 25 common frames omitted
Caused by: org.apache.ibatis.exceptions.PersistenceException: 
### Error querying database.  Cause: org.postgresql.util.PSQLException: ERROR: relation "act_ge_property" does not exist
  Position: 15
### The error may exist in org/camunda/bpm/engine/impl/mapping/entity/Property.xml
### The error may involve org.camunda.bpm.engine.impl.persistence.entity.PropertyEntity.selectProperty-Inline
### The error occurred while setting parameters
### SQL: select * from ACT_GE_PROPERTY where NAME_ = ?
### Cause: org.postgresql.util.PSQLException: ERROR: relation "act_ge_property" does not exist
  Position: 15
        at org.apache.ibatis.exceptions.ExceptionFactory.wrapException(ExceptionFactory.java:30)
        at org.apache.ibatis.session.defaults.DefaultSqlSession.selectList(DefaultSqlSession.java:149)
        at org.apache.ibatis.session.defaults.DefaultSqlSession.selectList(DefaultSqlSession.java:140)
        at org.apache.ibatis.session.defaults.DefaultSqlSession.selectOne(DefaultSqlSession.java:76)
        at org.camunda.bpm.engine.impl.db.sql.DbSqlSession.lambda$selectById$2(DbSqlSession.java:120)
        at org.camunda.bpm.engine.impl.util.ExceptionUtil.doWithExceptionWrapper(ExceptionUtil.java:254)
        ... 77 common frames omitted
Caused by: org.postgresql.util.PSQLException: ERROR: relation "act_ge_property" does not exist
  Position: 15
        at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2674)
        at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2364)
        at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:354)
        at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:484)
        at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:404)
        at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:162)
        at org.postgresql.jdbc.PgPreparedStatement.execute(PgPreparedStatement.java:151)
        at com.zaxxer.hikari.pool.ProxyPreparedStatement.execute(ProxyPreparedStatement.java:44)
        at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.execute(HikariProxyPreparedStatement.java)
        at org.apache.ibatis.executor.statement.PreparedStatementHandler.query(PreparedStatementHandler.java:64)
        at org.apache.ibatis.executor.statement.RoutingStatementHandler.query(RoutingStatementHandler.java:79)
        at org.apache.ibatis.executor.BatchExecutor.doQuery(BatchExecutor.java:92)
        at org.apache.ibatis.executor.BaseExecutor.queryFromDatabase(BaseExecutor.java:325)
        at org.apache.ibatis.executor.BaseExecutor.query(BaseExecutor.java:156)
        at org.apache.ibatis.executor.CachingExecutor.query(CachingExecutor.java:109)
        at org.apache.ibatis.executor.CachingExecutor.query(CachingExecutor.java:89)
        at org.apache.ibatis.session.defaults.DefaultSqlSession.selectList(DefaultSqlSession.java:147)
        ... 81 common frames omitted
```

--- a/micronaut-camunda-bpm-feature/build.gradle
+++ b/micronaut-camunda-bpm-feature/build.gradle
@@ -21,6 +21,7 @@ micronaut {
 dependencies {
     implementation("io.micronaut:micronaut-validation")
     implementation("io.micronaut:micronaut-runtime")
+    implementation("com.fasterxml.uuid:java-uuid-generator:4.0.1")
     api("org.camunda.bpm:camunda-engine:$camundaVersion") {
         exclude group: 'org.camunda.bpm.model', module: 'camunda-cmmn-model'
     }

--- a/micronaut-camunda-bpm-feature/src/main/java/info/novatec/micronaut/camunda/bpm/feature/IdGeneratorFactory.java
+++ b/micronaut-camunda-bpm-feature/src/main/java/info/novatec/micronaut/camunda/bpm/feature/IdGeneratorFactory.java
@@ -1,0 +1,34 @@
+/*
+ * Copyright 2022 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package info.novatec.micronaut.camunda.bpm.feature;
+
+import io.micronaut.context.annotation.Factory;
+import jakarta.inject.Singleton;
+import org.camunda.bpm.engine.impl.cfg.IdGenerator;
+import org.camunda.bpm.engine.impl.persistence.StrongUuidGenerator;
+
+/**
+ * @author Tobias Schäfer
+ */
+// Implementation based on https://github.com/camunda/camunda-bpm-spring-boot-starter/blob/master/starter/src/main/java/org/camunda/bpm/spring/boot/starter/configuration/id/IdGeneratorConfiguration.java
+@Factory
+public class IdGeneratorFactory {
+
+    @Singleton
+    public IdGenerator idGenerator() {
+        return new StrongUuidGenerator();
+    }
+}

--- a/micronaut-camunda-bpm-feature/src/main/java/info/novatec/micronaut/camunda/bpm/feature/MnProcessEngineConfiguration.java
+++ b/micronaut-camunda-bpm-feature/src/main/java/info/novatec/micronaut/camunda/bpm/feature/MnProcessEngineConfiguration.java
@@ -206,6 +206,7 @@ public class MnProcessEngineConfiguration extends ProcessEngineConfigurationImpl
                 "        <mapper resource=\"org/camunda/bpm/engine/impl/mapping/entity/Tenant.xml\" />\n" +
 
                 "        <mapper resource=\"org/camunda/bpm/engine/impl/mapping/entity/Deployment.xml\" />\n" +
+                "        <mapper resource=\"org/camunda/bpm/engine/impl/mapping/entity/Incident.xml\" />\n" + // e.g. New process definition is deployed which replaces a previous version that included a Timer Start Event
                 "        <mapper resource=\"org/camunda/bpm/engine/impl/mapping/entity/Job.xml\" />\n" +
                 "        <mapper resource=\"org/camunda/bpm/engine/impl/mapping/entity/JobDefinition.xml\" />\n" +
                 "        <mapper resource=\"org/camunda/bpm/engine/impl/mapping/entity/ProcessDefinition.xml\" />\n" +
@@ -259,7 +260,7 @@ public class MnProcessEngineConfiguration extends ProcessEngineConfigurationImpl
                 "    <mapper resource=\"org/camunda/bpm/engine/impl/mapping/entity/IdentityLink.xml\" />\n" +
                 //"    <mapper resource=\"org/camunda/bpm/engine/impl/mapping/entity/Job.xml\" />\n" +
                 //"    <mapper resource=\"org/camunda/bpm/engine/impl/mapping/entity/JobDefinition.xml\" />\n" +
-                "    <mapper resource=\"org/camunda/bpm/engine/impl/mapping/entity/Incident.xml\" />\n" +
+                //"    <mapper resource=\"org/camunda/bpm/engine/impl/mapping/entity/Incident.xml\" />\n" +
                 //"    <mapper resource=\"org/camunda/bpm/engine/impl/mapping/entity/Membership.xml\" />\n" +
                 //"    <mapper resource=\"org/camunda/bpm/engine/impl/mapping/entity/ProcessDefinition.xml\" />\n" +
                 //"    <mapper resource=\"org/camunda/bpm/engine/impl/mapping/entity/Property.xml\" />\n" +

--- a/micronaut-camunda-bpm-feature/src/main/java/info/novatec/micronaut/camunda/bpm/feature/MnProcessEngineConfiguration.java
+++ b/micronaut-camunda-bpm-feature/src/main/java/info/novatec/micronaut/camunda/bpm/feature/MnProcessEngineConfiguration.java
@@ -28,6 +28,7 @@ import org.apache.ibatis.builder.xml.XMLConfigBuilder;
 import org.camunda.bpm.engine.ProcessEngineConfiguration;
 import org.camunda.bpm.engine.history.*;
 import org.camunda.bpm.engine.impl.*;
+import org.camunda.bpm.engine.impl.cfg.IdGenerator;
 import org.camunda.bpm.engine.impl.cfg.ProcessEngineConfigurationImpl;
 import org.camunda.bpm.engine.impl.cfg.ProcessEnginePlugin;
 import org.camunda.bpm.engine.impl.cmmn.CaseServiceImpl;
@@ -93,6 +94,7 @@ public class MnProcessEngineConfiguration extends ProcessEngineConfigurationImpl
                                         CamundaVersion camundaVersion,
                                         ApplicationContext applicationContext,
                                         DataSource dataSource,
+                                        IdGenerator idGenerator,
                                         MnArtifactFactory artifactFactory,
                                         MnBeansResolverFactory beansResolverFactory,
                                         List<ProcessEnginePlugin> plugins,
@@ -109,6 +111,7 @@ public class MnProcessEngineConfiguration extends ProcessEngineConfigurationImpl
         mockUnsupportedCmmnMethods();
         setDataSource(dataSource);
         setTransactionsExternallyManaged(true);
+        setIdGenerator(idGenerator);
         setExpressionManager(new MnExpressionManager(new ApplicationContextElResolver(applicationContext)));
         setArtifactFactory(artifactFactory);
 

--- a/micronaut-camunda-bpm-feature/src/test/kotlin/info/novatec/micronaut/camunda/bpm/feature/test/ProcessEngineFactoryDeploymentTest.kt
+++ b/micronaut-camunda-bpm-feature/src/test/kotlin/info/novatec/micronaut/camunda/bpm/feature/test/ProcessEngineFactoryDeploymentTest.kt
@@ -20,41 +20,67 @@ import io.micronaut.context.annotation.Property
 import io.micronaut.test.extensions.junit5.annotation.MicronautTest
 import jakarta.inject.Inject
 import org.assertj.core.api.Assertions.assertThat
+import org.camunda.bpm.engine.ManagementService
 import org.camunda.bpm.engine.RepositoryService
 import org.camunda.bpm.engine.repository.Deployment
 import org.camunda.bpm.engine.repository.ProcessDefinition
+import org.camunda.bpm.engine.repository.Resource
+import org.junit.jupiter.api.Nested
 import org.junit.jupiter.api.Test
+import java.util.stream.Collectors
 
-@MicronautTest
-@Property(name="camunda.locations", value="classpath:.,classpath:bpm")
 class ProcessEngineFactoryDeploymentTest {
 
-    @Inject
-    lateinit var repositoryService: RepositoryService
+    @MicronautTest
+    @Nested
+    inner class SimpleDeployment {
 
-    @Test
-    fun `deployment name is set`() {
-        assertThat(repositoryService.createDeploymentQuery().list())
-            .extracting<String>(Deployment::getName)
-            .contains(DeployModelsExecutor.MICRONAUT_AUTO_DEPLOYMENT_NAME)
+        @Inject
+        lateinit var repositoryService: RepositoryService
+
+        @Inject
+        lateinit var managementService: ManagementService
+
+        @Test
+        fun `deployment name is set`() {
+            assertThat(repositoryService.createDeploymentQuery().list())
+                .extracting<String>(Deployment::getName)
+                .contains(DeployModelsExecutor.MICRONAUT_AUTO_DEPLOYMENT_NAME)
+        }
+
+        @Test
+        fun `camunda forms get deployed`() {
+            assertThat(managementService.registeredDeployments.stream()
+                .map { deploymentId -> repositoryService.getDeploymentResources(deploymentId) }
+                .flatMap { obj -> obj.stream() }
+                .map(Resource::getName)
+                .collect(Collectors.toList())).contains("sample.form")
+        }
+
+        @Test
+        fun `DMN tables get deployed`() {
+            assertThat(managementService.registeredDeployments.stream()
+                .map { deploymentId -> repositoryService.getDeploymentResources(deploymentId) }
+                .flatMap { obj -> obj.stream() }
+                .map(Resource::getName)
+                .collect(Collectors.toList())).contains("simple-table.dmn")
+        }
     }
 
-    @Test
-    fun `all given locations are scanned`() {
-        assertThat(repositoryService.createProcessDefinitionQuery().latestVersion().list())
-            .extracting<String>(ProcessDefinition::getKey)
-            .contains("ProcessEmpty", "ProcessEmptySubdir")
-    }
+    @MicronautTest
+    @Property(name="camunda.locations", value="classpath:.,classpath:bpm")
+    @Nested
+    inner class Subdirectory {
 
-    @Test
-    fun `camunda forms get deployed`() {
-        assertThat(repositoryService.getDeploymentResourceNames("1"))
-            .contains("sample.form")
-    }
+        @Inject
+        lateinit var repositoryService: RepositoryService
 
-    @Test
-    fun `DMN tables get deployed`() {
-        assertThat(repositoryService.getDeploymentResourceNames("1"))
-            .contains("simple-table.dmn")
+        @Test
+        fun `all given locations are scanned`() {
+            assertThat(repositoryService.createProcessDefinitionQuery().latestVersion().list())
+                .extracting<String>(ProcessDefinition::getKey)
+                .contains("ProcessEmpty", "ProcessEmptySubdir")
+        }
+
     }
 }

